cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTask.java

```diff
@@ -65,16 +71,25 @@
    *
    * @param executionId The execution id of the proposal so we can keep track of the task when execute it.
    * @param proposal The corresponding balancing proposal of this task.
+   * @param brokerId The broker to operate on if the task is of type {@link TaskType#INTRA_BROKER_REPLICA_ACTION}.
    * @param type the {@link TaskType} of this task.
    */
-  public ExecutionTask(long executionId, ExecutionProposal proposal, TaskType type) {
+  public ExecutionTask(long executionId, ExecutionProposal proposal, Integer brokerId, TaskType type) {
```

Comment:
This is because one `ExecutionProposal` can generate multiple `ExecutionTask` for  intra-broker replica movement. it is not 1-to-1 mapping(which is in inter-broker replica movement case)

 And the root reason  is because for each of  intra-broker replica movements of the same partition, we need to send a `AlterReplicaLogdir` request to that broker